File: examples/scenarios/interrupted-flow-recovery/snippets/ios/trigger-cancel-event.swift
// Example placement: success/completion handler for campaign cancellation (pseudo-code).
// Adapt to: your app architecture, notification integration, and actual OpenClix runtime API names.
// Shared config: examples/scenarios/interrupted-flow-recovery/openclix.config.json
// Events used: critical_flow_completed, critical_flow_restarted
// Note: OpenClix APIs shown here are placeholders until the reference SDK is published.

import Foundation

func triggerinterruptedFlowRecoveryCancel(openClix: OpenClixManagerLike) {
  let event = OpenClixEventLike(name: "critical_flow_completed", appState: .foreground)
  openClix.ingestEvent(event)
  // Alternate cancel path in some apps: critical_flow_restarted
}

File: examples/scenarios/activation-first-value-rescue/snippets/react-native/triggerCancelEvent.ts
// Example placement: success/completion handler for campaign cancellation in JS/TS layer (pseudo-code).
// Adapt to: your app architecture, notification integration, and actual OpenClix runtime API names.
// Shared config: examples/scenarios/activation-first-value-rescue/openclix.config.json
// Events used: first_value_completed, onboarding_step_viewed (optional support event, not required for config)
// Note: OpenClix APIs shown here are placeholders until the reference SDK is published.

type OpenClixManagerLike = {
  ingestEvent: (event: { name: string; appState: 'foreground' | 'background' }) => void;
};

export function triggerActivationFirstValueRescueCancel(openClix: OpenClixManagerLike): void {
  openClix.ingestEvent({
    name: 'first_value_completed',
    appState: 'foreground',
  });
  // Alternate cancel path in some apps: onboarding_step_viewed (optional support event, not required for config)
}
